datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -435,13 +427,16 @@ macro_rules! slice {
             .zip($EXTRA_KEY.iter())
             .map(|((arr, i), j)| match (arr, i, j) {
                 (Some(arr), Some(i), Some(j)) => {
-                    list_slice!(arr, i, j, $RETURN_ELEMENT, $ARRAY_TYPE)
+                    let arr = arr.as_any().downcast_ref::<$ARRAY_TYPE>().unwrap();
```

Comment:
since `list_slice` takes a `array: &dyn Array,` already, is it possible to avoid the `arr.as_any().downcast_ref...` call and just pass `arr` directly to `list_slice`?